parse/functions.go

```diff
@@ -563,15 +563,32 @@ var (
 		},
 		"sum": {
 			ValidateArgs: func(args []*types.DataType) (*types.DataType, error) {
+				// per https://www.postgresql.org/docs/current/datatype-numeric.html#DATATYPE-NUMERIC-TABLE
+				// the result of sum will be made a decimal(1000, 0)
 				if len(args) != 1 {
 					return nil, wrapErrArgumentNumber(1, len(args))
 				}
 
-				if !args[0].EqualsStrict(types.IntType) {
-					return nil, wrapErrArgumentType(types.IntType, args[0])
+				if !args[0].IsNumeric() {
+					return nil, fmt.Errorf("expected argument to be numeric, got %s", args[0].String())
 				}
 
-				return types.IntType, nil
+				var retType *types.DataType
+				switch {
+				case args[0].EqualsStrict(types.IntType):
+					retType = decimal1000.Copy()
+				case args[0].Name == types.DecimalStr:
+					retType = args[0].Copy()
+					retType.Metadata[0] = 1000 // max precision
+				case args[0].EqualsStrict(types.Uint256Type):
+					retType = decimal1000.Copy()
+				case args[0].EqualsStrict(types.UnknownType):
+					retType = types.UnknownType.Copy()
```

Comment:
`EqualsStrict` has this:

```go
	// if unknown, return true. unknown is a special case used
	// internally when type checking is disabled.
	if c.Name == unknownStr || other.Name == unknownStr {
		return true
	}
```

So this can be true with any type?

Reply:
Oh jeez, that's a good catch. Will fix this